src/types/dictionaryTypes.ts

```diff
@@ -240,5 +292,62 @@ export const Dictionary = zod
 		(dictionary) => allUnique(dictionary.schemas.map((schema) => schema.name)),
 		// TODO: Can improve uniqueness check error by providing list of duplicate field names.
 		'All schemas in the dictionary must have a unique name.',
-	);
+	)
+	.superRefine((dictionary, ctx) => {
+		/**
+		 * Enforce schema foreignKey restrictions:
+		 * 1. make sure that the foreign schema referenced in all provided foreignKey restrictions have matching schemas in the dicitonary
+		 * 2. make sure all foreign fields listed in foreignKey restrictions have matching fields in the specified foreign schemas
+		 */
+
+		// List of all schemas and all their fields, for cross reference from the foreignKey checks
+		const schemaNames = dictionary.schemas.map((schema) => schema.name);
+		const schemaFieldMap: Record<string, string[]> = dictionary.schemas.reduce<Record<string, string[]>>(
+			(acc, schema) => {
+				const fieldNames = schema.fields.map((field) => field.name);
+				return { ...acc, [schema.name]: fieldNames };
+			},
+			{},
+		);
+
+		// Loop through each schema and apply checks if they have a foreignKey restriction
+		dictionary.schemas.forEach((schema) => {
+			if (schema.restrictions?.foreignKey) {
+				// We have a foreign key restriction, now we will now check its properties all reference existing schemas and fields
+
+				schema.restrictions.foreignKey.forEach((fkRestriction) => {
+					if (!schemaNames.includes(fkRestriction.schema)) {
```

Comment:
While I'd truly prefer you didn't use negative logic in general, in this particular case the extra cognitive burden is not just avoidable but unnecessary as you can just apply the opposite if.

Reply:
Fair, I'll swap the order.